core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -513,13 +516,44 @@ private[spark] object Utils extends Logging {
       Files.move(sourceFile, destFile)
```

Comment:
It's not clear from the docs, but does this work for directories? Could you add a unit test? (It doesn't seem like this is covered for directories by the current ones, after a quick look.)